src/updater.js

```diff
@@ -0,0 +1,55 @@
+const { dialog } = require('electron');
+const { autoUpdater } = require('electron-updater');
+const noUpdateDialog = require('./no_update_dialog.js');
+const updateAvailableDialog = require('./update_available_dialog.js');
+
+// Pointer to menuItem so we can manage it's state.
+let updateMenuItem = null;
+// If menu is clicked on we'll set the event. Used to decide about showing dialog.
+let updateEvent = null;
+
+let updateInfo = null;
+
+autoUpdater.autoDownload = false;
+
+autoUpdater.on('error', (error) => {
+  // TODO: This needs ux for a better user experience.
+  dialog.showErrorBox('Error: ', error == null ? "unknown" : (error.stack || error).toString());
+  // Re-enable menuItem since something went wrong while checking.
+  updateMenuItem.enabled = true;
+})
+
+autoUpdater.on('update-available', (updateInfo) => {
+  updateInfo = updateInfo;
+  updateAvailableDialog.open(updateInfo);
+  updateMenuItem.enabled = true;
```

Comment:
In this and other event handlers,  we're enabling the 'check for updates' menu when we show the update available/not available.  This is because the check is complete and we're re-allowing the menu? 

Would it make more sense to have the owner of the menu item control its enabled state? Or does that become cumbersome?

Reply:
The separation of concerns here is something I was trying to fix this weekend but failed. I think as we take it a bit further in the next few days the pattern will show itself. I went with simplest to get it working.